library/objective-c/EnvoyNetworkMonitor.m

```diff
@@ -38,6 +43,7 @@ - (void)startPathMonitor {
   nw_path_monitor_set_queue(_path_monitor, queue);
 
   __block envoy_network_t previousNetworkType = -1;
+  envoy_engine_t engineHandle = _engineHandle;
```

Comment:
Why was this needed?

Reply:
It was causing a retain cycle in the nw update block.